rust/perspective-viewer/src/rust/components/number_column_style.rs

```diff
@@ -237,34 +259,41 @@ impl Component for NumberColumnStyle {
             NumberColumnStyleMsg::GradientChanged(side, gradient) => {
                 match (side, gradient.parse::<f64>()) {
                     (Fg, Ok(x)) => {
-                        self.fg_gradient = x;
+                        self.fg_gradient = Some(x);
                         self.config.fg_gradient = Some(x);
                     }
                     (Fg, Err(_)) if gradient.is_empty() => {
-                        self.fg_gradient = self.default_config.fg_gradient;
+                        self.fg_gradient = Some(self.default_config.fg_gradient);
                         self.config.fg_gradient = Some(self.default_config.fg_gradient);
                     }
                     (Fg, Err(_)) => {
-                        self.fg_gradient = self.default_config.fg_gradient;
+                        self.fg_gradient = Some(self.default_config.fg_gradient);
                         self.config.fg_gradient = None;
                     }
                     (Bg, Ok(x)) => {
-                        self.bg_gradient = x;
+                        self.bg_gradient = Some(x);
                         self.config.bg_gradient = Some(x);
                     }
                     (Bg, Err(_)) if gradient.is_empty() => {
-                        self.bg_gradient = self.default_config.bg_gradient;
+                        self.bg_gradient = Some(self.default_config.bg_gradient);
                         self.config.bg_gradient = Some(self.default_config.bg_gradient);
                     }
                     (Bg, Err(_)) => {
-                        self.bg_gradient = self.default_config.bg_gradient;
+                        self.bg_gradient = Some(self.default_config.bg_gradient);
                         self.config.bg_gradient = None;
                     }
                 };
 
                 self.dispatch_config(ctx);
                 false
             }
+            NumberColumnStyleMsg::DefaultGradientChanged(gradient) => {
```

Comment:
Distributing the default value logic across multiple files like means we have the of the complexity of the `default_config`/`config` design, but none of the decoupling.  We should instead IMO make the `default_config()` getter an async method which can query it's container.